### PR TITLE
Bot API 6.8: Stories, voting as a chat, and other minor changes

### PR DESCRIPTION
**Pull request by coder2020official**
### **Bot API 6.8**

- [x] Added the field story to the class [Message](https://core.telegram.org/bots/api#message) for messages with forwarded stories. Currently, it holds no information.

- [x] Added the field voter_chat to the class [PollAnswer](https://core.telegram.org/bots/api#pollanswer), to contain channel chat voters in [Polls](https://core.telegram.org/bots/api#poll). For backward compatibility, the field user in such objects will contain the user 136817688 ([@Channel_Bot](https://t.me/Channel_Bot)).
- [x] Added the field emoji_status_expiration_date to the class [Chat](https://core.telegram.org/bots/api#chat).
- [x] Added the method [unpinAllGeneralForumTopicMessages](https://core.telegram.org/bots/api#unpinallgeneralforumtopicmessages).
- [x] Increased to 512 characters the maximum length of the startapp parameter in direct Web App **links(NOT NEEDED)**

### CODE REVIEW

**Comment by coder2020official** [comment]
@badiboy can you clarify one thing: what is the purpose of to_dict in types.PollAnswer? Check previous commit where I added voter_chat, and if to_dict is necessary I have to add to_dict for types.Chat. I have checked api doc and I see no need for to_dict for PollAnswer, since it is an update. Or did I miss something?

**Comment by coder2020official** [comment]
I guess I am done except for the PollAnswer issue I described above.

**Comment by Badiboy** [comment]
> @Badiboy can you clarify one thing: what is the purpose of to_dict in types.PollAnswer?

It was long long time ago. :)
https://github.com/eternnoir/pyTelegramBotAPI/pull/839

They are not needed in library. And I suppose they are not used by anybody.
I prefer them to stay as far as they are already exists, but if you **hardly** want to remove it - I think they can be removed.

**Comment by Badiboy** [comment]
The same you can check user.to_dict/to_json if you want to, I do not remember what's with these methods usage.

File: telebot/types.py [hunk 1]
```diff
@@ -7743,3 +7758,18 @@ def to_dict(self) -> dict:
...
+
+class Story(JsonDeserializable):
+    """
+    This object represents a message about a forwarded story in the chat.
+    Currently holds no information.
+    """
+
+    @classmethod
+    def de_json(cls, json_string):
+        return cls()
```
Comment (Badiboy) [inline telebot/types.py:7783]: image

Generally we proceed json in this way. Currently it does not matter, but I suppose it's better to have standart code signature even here.

File: telebot/types.py [hunk 2]
```diff
@@ -1274,10 +1280,19 @@ def __init__(self, message_id, from_user, date, chat, content_type, options, jso
         self.write_access_allowed: Optional[WriteAccessAllowed] = None
         self.user_shared: Optional[UserShared] = None
         self.chat_shared: Optional[ChatShared] = None
+        self.story: Optional[Story] = None
         for key in options:
             setattr(self, key, options[key])
         self.json = json_string
 
+    @property
+    def is_forwarded_story(self):
```
Comment (Badiboy) [inline telebot/types.py:1294]: See no reason in appearance of this property... No is_anumation, is_video, is_xxx - why now is_story?

File: telebot/types.py [hunk 3]
```diff
@@ -6710,12 +6713,16 @@ def de_json(cls, json_string):
         if (json_string is None): return None
         obj = cls.check_json(json_string)
         obj['user'] = User.de_json(obj['user'])
+        if 'voter_chat' in obj:
```
Comment (Badiboy) [inline telebot/types.py:6721]: User is now optional and should be processed the same as voter_chat with if

File: telebot/types.py [hunk 4]
```diff
@@ -6710,12 +6713,16 @@ def de_json(cls, json_string):
...
+            obj['voter_chat'] = Chat.de_json(obj['voter_chat'])
         return cls(**obj)
 
-    def __init__(self, poll_id, user, option_ids, **kwargs):
+    def __init__(self, poll_id, user, option_ids, voter_chat=None, **kwargs):
```
Comment (Badiboy) [inline telebot/types.py:6725]: And user is optional here also... Seems we need to change init signature.

File: telebot/types.py [hunk 5]
```diff
@@ -6710,12 +6713,16 @@ def de_json(cls, json_string):
...
         self.poll_id: str = poll_id
         self.user: User = user
```
Comment (Badiboy) [inline telebot/types.py:6727]: And here...

**Comment by Badiboy** [comment]
@coder2020official I submitted review. If you don't see it let me know, means I did something wrong again )

**Comment by coder2020official** [comment]
@badiboy i guess I am done, I removed the is_forwarded_story option(found it useless after ur comment), made user optional, improved new story class to check for json. Check the previous commit for more

**Comment by coder2020official** [comment]
Tip: You can also suggest a commit and I will be able to commit it right away
image

**Review by Badiboy — APPROVED** [review]

**Pull request merged**